llvm/include/llvm/IR/PatternMatch.h

```diff
@@ -843,9 +843,9 @@ struct bind_const_intval_ty {
 /// Match a specified integer value or vector of all elements of that
 /// value.
 template <bool AllowUndefs> struct specific_intval {
-  APInt Val;
+  const APInt &Val;
```

Comment:
Hm, is this safe for all usages? Might the APInt be temporary?

Reply:
It only needs to be alive through the m_SpecificInt call where the temporary would be created in the match expression